### PR TITLE
feat(favorite): #MED-239 add or remove favorite

### PR DESCRIPTION
**Pull request by Mattox0**
## Describe your changes

We can add or remove favorite resource

image

## Checklist tests

- Check if you can add favorite in textbooks and signets
- Check if you can remove favorite in textbooks and signets

## Issue ticket number and link

[https://jira.support-ent.fr/browse/MED-239](MED-239)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/resource/Resource.tsx [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+import { useTranslation } from "react-i18next";
 
 import { ListCardTypeEnum } from "~/core/enum/list-card-type.enum";
+import {
+  useAddFavoriteMutation,
+  useRemoveFavoriteMutation,
+} from "~/services/api/favorite.service";
 
 interface ResourceProps {
+  resource: any;
```
Comment (FlorentMr) [inline frontend/src/components/resource/Resource.tsx:19]: Possible de typer resource ou trop long / inutile?

File: frontend/src/hooks/useFavorite.ts [hunk 2]
```diff
@@ -8,8 +8,13 @@ export const useFavorite = () => {
   const [favorites, setFavorites] = useState<Favorite[]>([]);
 
   useEffect(() => {
-    setFavorites(favorite?.data ?? []);
+    let favoriteData = favorite?.data?.length > 0 ? favorite.data : [] ?? [];
```
Comment (FlorentMr) [inline frontend/src/hooks/useFavorite.ts:11]: typing possible?

**Review by FlorentMr — APPROVED** [review]

File: frontend/src/components/resource/Resource.tsx (hunk 1 above)
Comment (Mattox0) [reply frontend/src/components/resource/Resource.tsx:19]: Enfait une resource peut être autant un Signet, un Textbook, ou un Favoris, ça vaut le coup de typer comme ça ?

File: frontend/src/hooks/useSignet.ts [hunk 3]
```diff
@@ -21,25 +23,52 @@ export const useSignet = () => {
...
       (signet: Signet) => ({
         ...signet,
         orientation: signet.document_types.some((type) =>
           type.toLowerCase().includes("orientation"),
         ),
       }),
     );
-    setHomeSignets([...updatedPublicSignetsData, ...mySignetsData]);
-  }, [publicSignets, mySignets, user?.userId]);
+    let signetsData = [...updatedPublicSignetsData, ...mySignetsData];
```
Comment (alicedraillard) [inline frontend/src/hooks/useSignet.ts:42]: typage ?

File: frontend/src/routes/app/index.tsx [hunk 4]
```diff
@@ -30,32 +30,72 @@ export interface AppProps {
 export const App = () => {
   const [windowWidth, setWindowWidth] = useState(window.innerWidth);
   const [alertText, setAlertText] = useState<string>("");
-  const { favorites } = useFavorite();
-  const { homeSignets } = useSignet();
-  const { textbooks } = useTextbook();
+  const [alertType, setAlertType] = useState<AlertTypes>("success");
+  const [, forceUpdate] = useReducer((x) => x + 1, 0);
```
Comment (alicedraillard) [inline frontend/src/routes/app/index.tsx:34]: [, = typo ?

Comment (Mattox0) [reply frontend/src/routes/app/index.tsx:34]: le forceUpdate sert juste à re-render mon composant, je n'utilise donc pas le premier paramètre, il n'a pas trop de typo ducoup, si ?

**Review by FlorentMr — APPROVED** [review]

**Review by alicedraillard — APPROVED** [review]

**Pull request merged**